### PR TITLE
Removing hubstate

### PR DESCRIPTION
**Pull request by moozzyk**

### CODE REVIEW

File: test/Microsoft.AspNetCore.SignalR.CompatTests/ChatHubTests.cs [hunk 1]
```diff
@@ -58,44 +56,18 @@ public ChatHubTests(ServerFixture fixture)
 
         [Theory]
         [MemberData(nameof(AllTransports))]
-        public async Task ClientState_can_be_read_by_client(ITransportFactory transportFactory)
-        {
-            // NOTE: ClientState is probably going to be removed. However it is important to test how the managed client behaves when trying
```
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.SignalR.CompatTests/ChatHubTests.cs]: Is there a test that does what this note says?

Comment (moozzyk) [inline test/Microsoft.AspNetCore.SignalR.CompatTests/ChatHubTests.cs]: No - the change in behavior (@JunTaoLuo: behaviour) is not observable from the client. You will get the value you stored and you cannot change the value from the server since the API is gone. This is why I removed the test entirely.

Comment (JunTaoLuo) [inline test/Microsoft.AspNetCore.SignalR.CompatTests/ChatHubTests.cs]: On second thought I'll switch over eventually, when in America do as the Americans do.

**Comment by davidfowl** [comment]
👏

File: src/Microsoft.AspNetCore.SignalR.Server/Hubs/SignalProxy.cs [hunk 2]
```diff
@@ -9,11 +9,11 @@
 
 namespace Microsoft.AspNetCore.SignalR.Hubs
 {
-    public abstract class SignalProxy : DynamicObject, IClientProxy
+    public class SignalProxy : DynamicObject, IClientProxy
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.SignalR.Server/Hubs/SignalProxy.cs]: Why this change?

Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.SignalR.Server/Hubs/SignalProxy.cs]: Probably cause `StatefulSignalProxy` was the only class inheriting from this

Comment (moozzyk) [inline src/Microsoft.AspNetCore.SignalR.Server/Hubs/SignalProxy.cs]: What @BrennanConroy said. After removing unneeded members from StatefulSignalProxy there was nothing left so I removed the entire class. But it was derived from SignalProxy which was abstract so I had to make it non-abstract to be able to create instances in places where we used to create StatefulSignalProxy

File: test/Microsoft.AspNetCore.SignalR.CompatTests/ChatHubTests.cs (hunk 1 above)
Comment (moozzyk) [inline test/Microsoft.AspNetCore.SignalR.CompatTests/ChatHubTests.cs]: @JunTaoLuo - the biggest issue for me was Outlook - first it would autocorrect me and when I rejected these "corrections" it would put squiggles. I relented.

**Comment by BrennanConroy** [comment]
:shipit:
